### DISCUSSION TITLE
In operation uv__io_poll, value timeout is -1

### DISCUSSION THREAD

**Discussion by dSalieri** [libuv/libuv#4510] [Q&A] [answered]
The question reflects the title of the heading.

I read the source code of the library, but I don't quite understand what the lines [like that](https://github.com/nodejs/node/blob/main/deps/uv/src/unix/posix-poll.c#L247), mean:
```c
if (timeout == -1) continue;
```
Please explain what is this case when `timeout` is `-1`. And can timeout have a higher negative number?

**Comment by bnoordhuis** [comment, marked as answer]
You mean a _lower_ negative number? :-)

-1 means wait indefinitely, 0 means don't wait (don't suspend), and anything > 0 means "wait this many milliseconds" - unless events come in first, of course.

Anything < -1 is undefined.

**Reply by dSalieri** [reply]
@bnoordhuis  How does the value -1 come about? It's not entirely clear from the code. Could you please clarify?

About the lower negative number than -1. Can we get the larger negative value when we execute code related to polling events?

**Reply by bnoordhuis** [reply]
It's just a sentinel value. In a language like rust, I would've used an enum but C doesn't have those (as in: it has enums but not union type enums) so sentinel it is.

Libuv doesn't use values < -1.
